features/prevent_abuse.feature

```diff
@@ -0,0 +1,39 @@
+Feature: Safeguards to prevent abuse of self-hosted runners
+
+    Compute resources should be protected from abuse by malicious actors.
+    This includes preventing forks from accessing our resources and includes
+    preventing runs on untrusted PRs.
+
+    Scenario: Forks should not be able to use our runners
+        # This should be guaranteed by the fact that secrets don't propagate
+        # to forks.
+        Given a malicious actor's fork of our repository
```

Comment:
I think this should not be limited to a malicious actor. This should be the case regardless of if the user is malicious or not.

Reply:
Good call. Updated so that the intent isn't an explicit part of the story. Thinking from the perspective of someone with malicious intent is important to coming up with scenarios, but of course the software can't read the person's state of mind, so its behavior doesn't depend on that.